### PR TITLE
[dep] update criterion, and fix lints

### PR DESCRIPTION
**Pull request by rexhoffman**
## Motivation

Update criterion and fix deprecation.

I've upgraded (but not tested) crypto/crypto/benches/noise.rs to benchmarkgroups.

Network authors should do the same for:

```
 network/benches/network_bench.rs
 network/benches/socket_bench.rs
```
 
 There didn't seem to be a complete one-to-one correspondence from ParameterizedBench to a configured BenchmarkGroup

### Have you read the [Contributing Guidelines on pull requests]

Y

## Test Plan

CI

## Related PRs

https://github.com/diem/diem/pull/7347

### CODE REVIEW

**Comment by rexhoffman** [comment]
Also given @runtian-zhou's effort to run benchmarks in CI, perhaps we should be using Iai:  https://bheisler.github.io/criterion.rs/book/iai/iai.html.

**Comment by metajack** [comment]
Note for others about the deprecation: https://github.com/bheisler/criterion.rs/blob/master/CHANGELOG.md#034---2021-01-24

**Review by metajack — APPROVED** [review]

File: network/benches/network_bench.rs [hunk 1]
```diff
@@ -7,6 +7,8 @@
 #![allow(clippy::trivially_copy_pass_by_ref)]
 // Allow writing 1 * KiB or 1 * MiB
 #![allow(clippy::identity_op)]
+// Criterion API has changed, TODO: Remove parameterized groups, and bench()
+#![allow(deprecated)]
 
 use bytes::Bytes;
 use criterion::{
```
Comment (gregnazario) [inline network/benches/network_bench.rs:14]: Okay cool, I was debating removing the benches altogether given I don't know if they're even up to date anymore.

**Review by gregnazario — APPROVED** [review]

**Pull request merged**